### PR TITLE
[Component] Add quota check in diagnose me component.

### PR DESCRIPTION
**Pull request by numerology**
This change adds quota check to diagnose me component.

An optional input of a list of quota entries is expected. Then, the component compare what's available with this input to see if there is any shortage.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3062)

### CODE REVIEW

**Comment by numerology** [comment]
@SinaChavoshi 
Sorry for the auto-lint changes : (

The functional change happens hefore line 96 in component.py

File: components/diagnostics/diagnose_me/component.py [hunk 1]
```diff
@@ -29,134 +33,192 @@ def run_diagnose_me(
               If set to HALT_ON_ERROR will case any error to raise an exception.
               This is intended to stop the data processing of a pipeline. Can set
               to False to only report Errors/Warnings.
+          project_id:
+              GCP project ID which is assumed to be accessible from the pod.
```
Comment (SinaChavoshi) [inline components/diagnostics/diagnose_me/component.py]: GCP project ID which is assumed to be the project under which current pod is executing.

File: components/diagnostics/diagnose_me/component.py [hunk 2]
```diff
@@ -29,134 +33,192 @@ def run_diagnose_me(
...
+
+  import sys
+  from kfp.cli.diagnose_me import gcp
+
+  config_error_observed = False
+
+  quota_list = gcp.get_gcp_configuration(
+      gcp.Commands.GET_QUOTAS, human_readable=False
+  )
+  quota_dict = {}  # Mapping from region to dict[metric, available]
```
Comment (SinaChavoshi) [inline components/diagnostics/diagnose_me/component.py]: May want to add a logic to handle the case were `gcp.get_gcp_configuration` errors out. 

```
if not quota_list.has_error:
  for region_quota ...
else:
 print(
        'Failed to retrieve Project quota with error  %s\n' %
        (quota_list.stderr),
        file=sys.stderr
    )
    config_error_observed = True
```

**Comment by SinaChavoshi** [comment]
Nice work! Apart from the comment on getconfig error  looks good. 
/lgtm

File: components/diagnostics/diagnose_me/component.py (hunk 1 above)
Comment (numerology) [reply components/diagnostics/diagnose_me/component.py]: Done.

File: components/diagnostics/diagnose_me/component.py (hunk 2 above)
Comment (numerology) [reply components/diagnostics/diagnose_me/component.py]: Nice catch. Thanks!

**Comment by numerology** [comment]
Thanks! @SinaChavoshi

**Pull request merged**